### PR TITLE
Adding "clear brwosing data" logic on terminate

### PR DESCRIPTION
**Pull request by codeimpl**
If the automatic deletion option is turned on, the data should be
deleted when user click terminate button.

Refs: #790

### CODE REVIEW

File: banana_cake/java/org/banana/cake/CakeToolbarManager.java [hunk 1]
```diff
@@ -170,6 +171,7 @@ public void openClearBrowsingDataPreference() {
     public void terminate() {
         ChromeTabbedActivity activity = getChromeTabbedActivity();
         if (activity == null) return;
+        BananaClearBrowsingData.get().clearBrowsingData();
```
Comment (jdragonbae) [inline banana_cake/java/org/banana/cake/CakeToolbarManager.java:174]: Shouldn't we do this when feature is enabled?
`if (ExtensionFeatures.isEnabled(FeatureName.AUTO_CLEAR_BROWSING_DATA))`

Comment (codeimpl) [reply banana_cake/java/org/banana/cake/CakeToolbarManager.java]: Yes, you are right. Plz review it again. thanks.

**Comment by codeimpl** [comment]
1) When user click, delete data then call terminate func.
2) Need to check synchronous routine instead of asynchronous one.

File: banana_cake/interfaces/java/org/banana/cake/interfaces/BananaClearBrowsingData.java [hunk 2]
```diff
@@ -5,10 +5,12 @@
...
 
+import org.chromium.chrome.browser.browsing_data.BrowsingDataBridge.OnClearBrowsingDataListener;
+
 public interface BananaClearBrowsingData {
     static BananaClearBrowsingData get() {
         return BananaInterfaceProvider.get(BananaClearBrowsingData.class);
     }
 
-    void clearBrowsingData();
+    void clearBrowsingData(OnClearBrowsingDataListener onClearBrowsingDataListener);
```
Comment (romandev) [inline banana_cake/interfaces/java/org/banana/cake/interfaces/BananaClearBrowsingData.java:15]: You don't need to expose the Chromium type to upper layer.
You should make the BananaOnClearBrowsingDataListener.

**Review by romandev — APPROVED** [review]
lgtm

**Pull request merged**